kapitan/inputs/helm.py

```diff
@@ -272,3 +282,44 @@ def load_chart(self):
             raise HelmTemplateError(error_message)
 
         return yaml.safe_load_all(output)
+
+
+def check_data_for_b64(yml_obj):
+    """
+    check for .data in kind: Secret / ConfigMap
+    replace b64 encoded strings
+    """
+    for item in yml_obj:
+        kind = item.get("kind", None)
+        if kind and (kind == "Secret" or kind == "ConfigMap"):
+            if item.get("data", None):
+                # Option 1:
+                # item["data"] = replace_b64_refs(item["data"])
+
+                # Option 2:
+                item["stringData"] = replace_b64_refs(item["data"])
```

Comment:
Yeah I think option 2 is best. We are mangling data anyway, so best to make it into valid content in stringData, as data requires its content to be base64 and a ref is certainly not valid base64.

Reply:
I wish it would solve the original problem. It turned out that the key `.stringData` is not allowed in `ConfigMap`s. So now we have a mixed solution: 
* For `Secret`s `.data` is replaced with `.stringData` (originally option 2). 
* For `ConfigMap`s after the compile-step the `.data` key is base64 - encoded again (originally option 1).

But again, we can clarify this later in the workgroup meeting.